### DISCUSSION TITLE
Why are you reading data from the response object and not from the request object

### DISCUSSION THREAD

**Discussion by IonelLupu** [uNetworking/uWebSockets.js#982] [Q&A] [answered]
In this [example](https://github.com/uNetworking/uWebSockets.js/blob/master/examples/JsonPost.js) you are reading the data from the response object. Shouldn't the data be read from the request object? 

Usually, the user makes a request from the client to the server and that request contains the data. Then, the server reads that data from the request. After that, the server does some work and uses the response object to send data back to the client.

Ins't it the case here too?

**Comment by uasan** [comment]
Then it’s even more strange, if `onData` is a response method, why can’t it be used in another microtask, answer - because in fact `onData` is a method of the request object, but for some reason it’s in the response object )

**Comment by uNetworkingAB** [comment, marked as answer]
Honestly, it's splitting hairs. Consider it a quirk kept for backwards compatibility.

**Comment by slkerndnme** [comment]
We read data from request to prépare tje response. Literal wtf

Le sam. 28 oct. 2023, 01:06, uNetworkingAB ***@***.***> a
écrit :

**Reply by uNetworkingAB** [reply]
It makes sense if you think of it like so:

"HttpRequest" in uws really is "HttpHead" and "HttpResponse" in uws really is "HttpRequest".

That is - uws is **not** designed like other Node.js "frameworks" because **uws is not, "a framework"** atop the given design of Node.js.

The same goes with onAborted. It too "does not belong" where it is, until you think of it like HttpHead and HttpRequest. Yes, this requires thinking outside of the Node.js universe.

That's why uws is fast, and not slow. Because it isn't a smooth brain replica of what everyone else is doing.
